### DISCUSSION TITLE
Posibilidad de importar scripts de otro repositorio que use dotly

### DISCUSSION THREAD

**Discussion by EduardoSimon** [CodelyTV/dotly#48] [Ideas]
He conocido Dotly hace poco, lo he utilizado para mis propios dotfiles y me parece maravilloso, muchas felicidades por la maravilla que habéis creado 🤪. Lo he comentado con un compañero de trabajo y le ha encantado la posibilidad de "estandarizar" los dotfiles. Nos gusta mucho compartir y comentar scriptillos que mejoren la productividad, y vimos dotly como la oportunidad perfecta para poder compartirlos y tener una base común. 

Abro la issue para saber si está en el roadmap y como poder contribuir a una feature de este calibre. En mi cabeza sería un comando del estilo:

`dot download EduardoSimon utils/foo`

Donde:

- **download** sería el comando para importar o descargar dotfiles. Entiendo que import no puede ser porque ya está en uso.
- **EduardoSimon** sería el usuario de Github del que leer su repositorio de dotfiles basado en dotly
- **utils/foo** ruta dentro de la carpeta scripts que queremos bajarnos.

El prerequisito para que este comando funcione sería que el usuario del que queremos importar el script lo tenga público.
Se podría acceder al raw del fichero y guardartelo en local. 

Si no se especifica nada se bajaría de su rama por defecto, pero en siguientes iteraciones se podría especificar la rama y/o commit del que obtener el script.

En caso de no especificar el script que queremos descargar, se puede utilizar fzf para listar los scripts disponibles a descargar.

**Comment by EduardoSimon** [comment]
Hey @rgomezcasas or @JavierCane, how would you envision this feature? Do you think is worth to implement? I have been thinking in an easier way that can lead to a full implementation later. It should be feasible to add a dotly config file and specify where to look up for sources of dot scripts. Manually, the user could clone as submodules dotfiles from other dotly users and add it to the config file.

**Comment by gtrabanco** [comment]
So you want a decentralised market, isn't it?

It sounds quite interesting :D 

Maybe this could be also a repo in github and you can search for scripts there in a webpage in github.io pages or similar... I like a little bit. It adds some complex but its quite cool as well 😝

**Reply by EduardoSimon** [reply]
I wasn't referring to something that big, at least for an mvp. Maybe using some kind of configuration file for repositories using dotly that you want to import scripts from and a utility that lets you find a script in any of those repos. 

It doesn't sound so complicated to me, but I would like in advance if a PR like would be welcome by the repo owners @rgomezcasas  and @JavierCane

**Reply by gtrabanco** [reply]
Yes, I am working in a very basic tool to download scripts from other repos which could be cool for the first stage but ending with a all of marketplace to provide sysadmin better life 😂

**Reply by gtrabanco** [reply]
Have a look on pending approval PR #104
